test/integration/action_execution.go

```diff
@@ -543,23 +631,34 @@ func waitForDeploymentToUpdateResource(client kubeclientset.Interface, dep *apps
 		}
 		switch changeType {
 		case REQUEST_SINGLE_CONTAINER:
```

Comment:
Validating this way (comparing individual values and specific methods eg. `validaIncrement()`, etc.) is unwieldy and prone to coding problems.
I would rather copy the pod template spec from the original deployment post its creation, pass it into `newResizeActionExecutionDTO()` and update the respective container resource values to the same ones (desired values) that are requested in actionExecutionDTO and simple use `reflect.DeepEqual()` to compare the `podSpecs` here.